codex-rs/core/src/config_types.rs

```diff
@@ -43,11 +38,15 @@ impl<'de> Deserialize<'de> for McpServerConfig {
     {
         #[derive(Deserialize)]
         struct RawMcpServerConfig {
-            command: String,
+            command: Option<String>,
             #[serde(default)]
-            args: Vec<String>,
+            args: Option<Vec<String>>,
             #[serde(default)]
-            env: Option<HashMap<String, String>>,
+            env: Option<Option<HashMap<String, String>>>,
+
+            url: Option<String>,
+            bearer_token: Option<String>,
```

Comment:
I don't love that we encourage storing this in plaintext, but I admit we don't have a great alternative for people just yet.

Reply:
We should consider https://docs.rs/secrecy/latest/secrecy/